--- gthumb.orig/data/gthumb.desktop.in
+++ gthumb/data/gthumb.desktop.in
@@ -3,7 +3,6 @@
 _GenericName=Image Viewer
 _Comment=View and organize your images
 Categories=GNOME;GTK;Graphics;Viewer;RasterGraphics;2DGraphics;Photography;
-Encoding=UTF-8
 Exec=gthumb %U
 Icon=gthumb
 MimeType=image/bmp;image/jpeg;image/gif;image/png;image/tiff;image/x-bmp;image/x-ico;image/x-png;image/x-pcx;image/x-tga;image/xpm;image/svg+xml;
